### PR TITLE
refactor(utils): move is mount read only function to common lib

### PR DESCRIPTION
**Pull request by ChanYiLin**
ref: https://github.com/longhorn/longhorn/issues/7394

Add mount related util function: IsMountPointReadOnly
This is used in share-manager and instance-manager to detect if the mount point becomes readonly.

Note:
1. Since go-common-libs had some breaking (function parameters not the same) changes in this PR
So when updating the version, we also need to fix some code to meet the changes in go-common-libs

e.g.
```
if _, err := nsexec.Execute(nil, "mount", opts, lhtypes.ExecuteDefaultTimeout); err != nil {}
```

2. since go-common-libs has been updated, 
    - `go-spdk-helper` is also needed to update 
    - `longhorn-spdk-engine` is also needed to update

cc @derekbit

### CODE REVIEW

**Comment by ChanYiLin** [comment]
it seems we should wait until https://github.com/longhorn/longhorn/issues/7578 finished
and `longhorn-instance-manager` has been updated with the latest lib version
`go-spdk-helper`, `go-common-libs`, `longhorn-spdk-engine`
```
pkg/proxy/snapshot.go:69:101: not enough arguments in call to c.EngineSnapshotCreate
	have (string, string)
	want (string, string, *api.SnapshotOptions)
```

cc @derekbit

**Comment by derekbit** [comment]
> EngineSnapshotCreate

cc @DamiaSan

File: go.mod [hunk 1]
```diff
@@ -2,17 +2,19 @@ module github.com/longhorn/longhorn-instance-manager
...
 	github.com/google/uuid v1.6.0
-	github.com/longhorn/backupstore v0.0.0-20240126140439-9c90a8728ca2
-	github.com/longhorn/go-common-libs v0.0.0-20240109042507-23627e6416b7
-	github.com/longhorn/go-spdk-helper v0.0.0-20240117135122-26f8acb2a13d
+	github.com/longhorn/backupstore v0.0.0-20240219094812-3a87ee02df77
+	github.com/longhorn/go-common-libs v0.0.0-20240219094750-e7176c332156
+	github.com/longhorn/go-spdk-helper v0.0.0-20240301101140-6eb6aa5fc09d
 	github.com/longhorn/longhorn-engine v1.6.0-dev-20240105.0.20240126141003-067f67803ee8
-	github.com/longhorn/longhorn-spdk-engine v0.0.0-20240123044045-c5f14845bd83
+	github.com/longhorn/longhorn-spdk-engine v0.0.0-20240226075747-9b8228c933d5
```
Comment (innobead) [inline go.mod:14]: Don't update `longhorn-spdk-engine` dep in this PR, so you should be able to move this PR forwards. Daminao will take care of that update in another PR.

**Review by innobead — COMMENTED** [review]
LGTM. Resolve the feedback, then we can merge this PR.

File: go.mod (hunk 1 above)
Comment (innobead) [reply go.mod:14]: cc @DamiaSan

**Review by innobead — APPROVED** [review]

**Comment by innobead** [comment]
@mergify backport v1.6.x v1.5.x

**Comment by ChanYiLin** [comment]
MANUALLY BACKPORT
v16x: https://github.com/longhorn/longhorn-instance-manager/pull/427
v15x: https://github.com/longhorn/longhorn-instance-manager/pull/428

**Pull request merged**